Proofreading-OV-Runtime (#11658)
openvinotoolkit/openvino: tools/pot/openvino/tools/pot/algorithms/quantization/default/README.md

# DefaultQuantization Algorithm {#pot_compression_algorithms_quantization_default_README}

> ## Introduction

Reviewer: ```suggestion
```

> DefaultQuantization Algorithm was designed to perform fast and (in many cases) accurate quantization. It does not have any control of accuracy metric but provides a lot of options that can be used to improve it.

Reviewer: ```suggestion
The DefaultQuantization Algorithm was designed to perform fast and accurate quantization. It does not offer direct control over the accuracy metric itself but provides many options that can be used to improve it.
```

### Mandatory parameters
- `"preset"` - preset which controls the quantization mode (symmetric and asymmetric). It can take two values:
    - `"performance"` (default) - stands for symmetric quantization of weights and activations. This is the most 
    efficient across all the HW.
    - `"mixed"` - symmetric quantization of weights and asymmetric quantization of activations. This mode can be useful

>     for quantization of NN, which has both negative and positive input values in quantizing operations, e.g. 

Author: ```suggestion
    for quantization of NN, which has both negative and positive input values in quantizing operations, for example 
```

    non-ReLU based CNN.  
- `"stat_subset_size"` - size of subset to calculate activations statistics used for quantization. The whole dataset 
is used if no parameter specified. It is recommended to use not less than 300 samples.

### Optional parameters
All other options should be considered as an advanced and require deep knowledge of the quantization process. Below
is an overall description of all possible parameters:
- `"model type"` - an optional parameter, required for additional patterns in the model. The default value is "None" ("Transformer" is only other supported value now).
- `"inplace_statistic"` - an optional parameter, required for change of collect statistics method. This parameter reduces the amount of memory consumed, but increases the calibration time.
- `"ignored"` - NN subgraphs which should be excluded from the optimization process: 

>     - `"scope"` - list of particular nodes to exclude.

Author: ```suggestion
    - `"scope"` - a list of particular nodes to exclude.
```

>     - `"operations"` - list of operation types to exclude (expressed in OpenVINO IR notation). This list consists of

Author: ```suggestion
    - `"operations"` - a list of operation types to exclude (expressed in OpenVINO IR notation). This list consists of
```

    the following tuples:
        - `"type"` - type of ignored operation.

>         - `"attributes"` - if attributes are defined they will be considered during the ignorance. They are defined by

Author: ```suggestion
        - `"attributes"` - if attributes are defined, they will be considered during the ignorance. They are defined by
```

        a dictionary of `"<NAME>": "<VALUE>"` pairs.

> - `"weights"` - this section manually defines quantization scheme for weights and the way to estimate the 

Author: ```suggestion
- `"weights"` - this section entails manual definition of quantization scheme for weights and the way to estimate the 
```

quantization range for that. It is worth noting that changing the quantization scheme may lead to inability to infer such
mode on the existing HW.
    - `"bits"` - bit-width, default is "8".

>     - `"mode"` - quantization mode (symmetric or asymmetric).

Author: ```suggestion
    - `"mode"` - a quantization mode (symmetric or asymmetric).
```

    - `"level_low"` - minimum level in the integer range to quantize. The default is "0" for unsigned range, and for signed "-2^(bit-1)". 

>     - `"level_high"` - maximum level in the integer range to quantize. The default is "2^bits-1" for unsigned range, and for signed "2^(bit-1)-1".

Author: ```suggestion
    - `"level_high"` - the maximum level in the integer range to quantize. The default is "2^bits-1" for an unsigned range, and "2^(bit-1)-1" for a signed one.
```

    - `"granularity"` - quantization scale granularity. Can take the following two values:
        - `"pertensor"` (default) - per-tensor quantization with one scale factor and zero-point.
        - `"perchannel"` - per-channel quantization with per-channel scale factor and zero-point.
    - `"range_estimator"` - this section describes parameters of range estimator that is used in MinMaxQuantization 
    method to get the quantization ranges and filter outliers based on the collected statistics. Below are the parameters 
    that can be modified to get better accuracy results:
        - `"max"` - parameters to estimate top border of quantizing floating-point range:

>             - `"type"` - type of the estimator: 

Author: ```suggestion
            - `"type"` - a type of the estimator: 
```

>                 - `"max"` (default) - estimates the maximum in the quantizing set of value.

Author: ```suggestion
                - `"max"` (default) - estimates the maximum in the quantizing set of a value.
```

>                 - `"quantile"` - estimates the quantile in the quantizing set of value.

Author: ```suggestion
                - `"quantile"` - estimates the quantile in the quantizing set of a value.
```

            - `"outlier_prob"` - outlier probability used in the "quantile" estimator.
        - `"min"` - parameters to estimate bottom border of quantizing floating-point range:

>             - `"type"` - type of the estimator: 

Author: ```suggestion
            - `"type"` - a type of the estimator: 
```

>                 - `"min"` (default) - estimates the minimum in the quantizing set of value.

Author: ```suggestion
                - `"min"` (default) - estimates the minimum in the quantizing set of a value.
```

>                 - `"quantile"` - estimates the quantile in the quantizing set of value.

Author: ```suggestion
                - `"quantile"` - estimates the quantile in the quantizing set of a value.
```

            - `"outlier_prob"` - outlier probability used in the "quantile" estimator.

> - `"activations"` - this section manually defines quantization scheme for activations and the way to estimate the 

Author: ```suggestion
- `"activations"` - this section describes quantization scheme for activations and the way to estimate the 
```

quantization range for that. As before, changing the quantization scheme may lead to inability to infer such
mode on the existing HW:

>     - `"bits"` - bit-width, default is "8".

Author: ```suggestion
    - `"bits"` - bit-width, the default value is "8".
```

    - `"mode"` - quantization mode (symmetric or asymmetric).

>     - `"level_low"` - minimum level in the integer range to quantize. The default is "0" for unsigned range, and for signed "-2^(bit-1)".

Author: ```suggestion
    - `"level_low"` - the minimum level in the integer range to quantize. The default is "0" for an unsigned range, and "-2^(bit-1)" for a signed one.
```

>     - `"level_high"` - maximum level in the integer range to quantize. The default is "2^bits-1" for unsigned range, and for signed "2^(bit-1)-1". 

Author: ```suggestion
    - `"level_high"` - the maximum level in the integer range to quantize. The default is "2^bits-1" for an unsigned range, and "2^(bit-1)-1" for a signed one. 
```

        - `"max"` - parameters to estimate top border of quantizing floating-point range:

>             - `"aggregator"` - type of the function used to aggregate statistics obtained with estimator 

Author: ```suggestion
            - `"aggregator"` - a type of the function used to aggregate statistics obtained with the estimator 
```

            over the calibration dataset to get a value of the top border:
                - `"mean"` (default) - aggregates mean value.
                - `"max"` - aggregates max value.
                - `"min"` - aggregates min value.
                - `"median"` - aggregates median value.
                - `"mean_no_outliers"` - aggregates mean value after removal of extreme quantiles.
                - `"median_no_outliers"` - aggregates median value after removal of extreme quantiles.
                - `"hl_estimator"` - Hodges-Lehmann filter based aggregator.

>             - `"type"` - type of the estimator:

Author: ```suggestion
            - `"type"` - a type of the estimator:
```

                - `"max"` (default) - estimates the maximum in the quantizing set of value.
                - `"quantile"` - estimates the quantile in the quantizing set of value.
            - `"outlier_prob"` - outlier probability used in the "quantile" estimator.
        - `"min"` - parameters to estimate bottom border of quantizing floating-point range:

>             - `"type"` - type of the estimator: 

Author: ```suggestion
            - `"type"` - a type of the estimator: 
```

* [Quantization of Image Classification model](https://docs.openvino.ai/latest/pot_configs_examples_README.html) 

A template and full specification for DefaultQuantization algorithm for POT command-line inferface:

> * [Template](https://github.com/openvinotoolkit/testrepo/blob/master/tools/pot/configs/default_quantization_template.json)

Reviewer: I've got an Idea. How about we simply put it here in a roll-out panel? It is not long and if we format it as code, it should fit in nicely!
